scripts/automatic_tag.go

```diff
@@ -0,0 +1,274 @@
+// It checks if the tag is part of the current target (if it is belonging to a service or to core),
+// checks if it is newer than the current latest tag stored in the map and if it is, updates latestTags and returns it
+func storeLatestTag(t *plumbing.Reference, latestTags map[string]string, target string) (map[string]string, error) {
+	tagName, _ := strings.CutPrefix(t.Name().String(), "refs/tags/")
+	splitTag := strings.Split(tagName, "/")
+
+	switch target {
+	case core:
+		if len(splitTag) != 2 || splitTag[0] != "core" {
+			return latestTags, nil
+		}
+
+		version := splitTag[1]
+		if semver.Prerelease(version) != "" {
+			return latestTags, nil
+		}
+
+		// invalid (or empty) semantic version are considered less than a valid one
+		if semver.Compare(latestTags["core"], version) == -1 {
+			latestTags["core"] = version
+		}
+	case allServices:
+		if len(splitTag) != 3 || splitTag[0] != "services" {
+			return latestTags, nil
+		}
+
+		service := splitTag[1]
+		version := splitTag[2]
+		if semver.Prerelease(version) != "" {
+			return latestTags, nil
+		}
+
+		// invalid (or empty) semantic version are considered less than a valid one
+		if semver.Compare(latestTags[service], version) == -1 {
+			latestTags[service] = version
+		}
+	default:
+		return nil, fmt.Errorf("target not supported in storeLatestTag, fix the script")
+	}
+	return latestTags, nil
+}
+
+// computeUpdatedTag returns the a new tag with the updated version for a specific service according to the update type and target
+// example: for service argus with version v0.1.1, target all-services and updateType minor, it returns services/argus/v0.2.0
+func computeUpdatedTag(service, version, updateType, target string) (string, error) {
+	canonicalVersion := semver.Canonical(version)
+	splitVersion := strings.Split(canonicalVersion, ".")
+	if len(splitVersion) != 3 {
+		return "", fmt.Errorf("invalid canonical version")
+	}
+
+	switch updateType {
+	case patch:
+		patchNumber, err := strconv.Atoi(splitVersion[2])
+		if err != nil {
+			return "", fmt.Errorf("couldnt convert patch number to int")
+		}
+		updatedPatchNumber := patchNumber + 1
+		splitVersion[2] = fmt.Sprint(updatedPatchNumber)
+	case minor:
+		minorNumber, err := strconv.Atoi(splitVersion[1])
+		if err != nil {
+			return "", fmt.Errorf("couldnt convert minor number to int")
+		}
+		updatedPatchNumber := minorNumber + 1
+		splitVersion[1] = fmt.Sprint(updatedPatchNumber)
+		splitVersion[2] = "0"
+	default:
+		return "", fmt.Errorf("update type not supported in version increment, fix the script")
+	}
+
+	updatedVersion := strings.Join(splitVersion, ".")
+
+	switch target {
+	case core:
+		if service != "core" {
+			return "", fmt.Errorf("%s target was provided but store latest tag from another service: %s", target, service)
+		}
+		return fmt.Sprintf("core/%s", updatedVersion), nil
+	case allServices:
+		return fmt.Sprintf("services/%s/%s", service, updatedVersion), nil
```

Comment:
With this current logic, aren't we creating a `service/core/vX.Y.Z` tag when the target is `allServices`?

Reply:
Probably `computeUpdatedTag` doesn't need to know about `target`, and this logic could be moved to `automaticTagUpdate`.

(remembering comment above regarding `service` -> `module`) This would leave this function signature as `func computeUpdatedTag(module, version, updateType string)`, just focusing on calculating the next tag for module `module` at version `version` given the `updateType`

Reply:
I agree, that makes sense

Reply:
> With this current logic, aren't we creating a service/core/vX.Y.Z tag when the target is allServices?
No, because `lastestTags` wouldn't have the `core` module

Reply:
Done